optional_platform.cpp

```diff
@@ -13,10 +13,6 @@ __attribute__((weak)) int logMessage(log_priority priority, const char *fmt, ...
 
 __attribute__((weak)) extern const bool HAS_HW_KEY2 = false;
```

Comment:
I think we should remove this too.

Reply:
IMO, this field is useless. current time this section only can enter `CAN_RESET` is true.
and these platforms can have the BF2 keys.

If you agree, I'll remove related codes.

Reply:
I'd argue that powersaves can probably reset the cart, but more than likely does not handle key2 on it own, which means that it'd have to be done in software. Was the main reason I argued for having a way to do software key2.

Reply:
(n.b. by remove I mean this weak definition)